pandas-stubs/core/frame.pyi

```diff
@@ -900,15 +902,35 @@ class DataFrame(NDFrame, OpsMixin):
         thresh: int | None = ...,
         subset: ListLikeU | Scalar | None = ...,
         inplace: _bool | None = ...,
+        ignore_index: _bool = ...,
     ) -> DataFrame | None: ...
+    @overload
     def drop_duplicates(
         self,
         subset: Hashable | Iterable[Hashable] | None = ...,
         *,
         keep: NaPosition | _bool = ...,
-        inplace: _bool = ...,
+        inplace: Literal[False] = ...,
         ignore_index: _bool = ...,
     ) -> DataFrame: ...
+    @overload
+    def drop_duplicates(
+        self,
+        subset: Hashable | Iterable[Hashable] | None = ...,
+        *,
+        keep: NaPosition | _bool = ...,
+        inplace: Literal[True] = ...,
+        ignore_index: _bool = ...,
+    ) -> None: ...
+    @overload
```

Comment:
Reverse these two overloads.  Pattern should look like what is in `fillna()`.  First you'd have `inplace: Literal[True]` (without the ellipses), then you'd have `Literal[False] = ...`